src/room.cpp

```diff
@@ -37,8 +37,6 @@ void Room::Destroy() {
         }
         server = nullptr;
     }
-
-    room_thread->join();
```

Comment:
I think since we close the room, receiving packets isn't needed any more, and the way I arranged it, allows the room/server to work without a mutex